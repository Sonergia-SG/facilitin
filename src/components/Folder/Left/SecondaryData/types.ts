import {
  FolderMOAString,
  FolderMOEString,
  FolderSiteString,
} from '../../../../store/reducer/entities/types';

export interface FormFieldList {
  label: string;
  value: string | null;
  values: Array<{
    key: string;
    label: string;
  }>;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'list';
}

export interface FormFieldDate {
  label: string;
  value: string | null;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'date';
}

export interface FormFieldTextBicRule {
  format: 'bic';
}

export interface FormFieldTextIbanRule {
  format: 'iban';
}

export interface FormFieldTextPhoneRule {
  format: 'phone';
}

export type FormFieldTextRules =
  | FormFieldTextPhoneRule
  | FormFieldTextBicRule
  | FormFieldTextIbanRule;

export interface FormFieldText {
  label: string;
  value: string | null;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'text';
  rules?: FormFieldTextRules;
}

export interface FormFieldNumber {
  label: string;
  value: string | null;
  key: keyof FolderMOAString | keyof FolderMOEString | keyof FolderSiteString;
  type: 'number';
}

export type FormFieldDef =
  | FormFieldDate
  | FormFieldText
  | FormFieldNumber
  | FormFieldList;

export interface FormSectionDef {
  label: string;
  type: 'section';
  fields: Array<FormFieldDef>;
}

export type FormDef = FormFieldDef | FormSectionDef;

export type GenericForms = Array<FormDef>;

export interface Forms {
  moa: GenericForms;
  moe: GenericForms;
  site: GenericForms;
}
