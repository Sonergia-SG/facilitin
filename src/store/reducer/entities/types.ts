import {
  LOGOUT,
  FOLDER_LOADED,
  LIST_LOADED,
  FOLDER_UPDATE_MOA_LOADED,
  FOLDER_UPDATE_CHECK_POINT_LOADING,
  FOLDER_UPDATE_CHECK_POINT_LOADED,
  FOLDER_UPDATE_CHECK_POINT_ERROR,
  COMMENTS_LIST_LOADED,
  FOLDER_FILE_LITIGE_LOADED,
} from '../../types';

export interface User {
  id_user: number;
  nom: string;
  prenom: string;
}

export interface Users {
  [index: number]: User;
}

export interface CommentSimple {
  id_log: number;
  id_user: number;
  date_log: string;
  message: string;
}

export interface Comment extends CommentSimple {
  user: number;
}

export interface CommentFull extends CommentSimple {
  user: User;
}

export interface Comments {
  [index: number]: Comment;
}

export type BooleanNumber = 0 | 1;

export interface SimpleFile {
  id_dp_file: number;
  id_file: number;
  id_dp_operation: number;
  id_dossierprime: number;
  is_devis: BooleanNumber;
  is_aat: BooleanNumber;
  is_facture: BooleanNumber;
  is_lettre_engagement: BooleanNumber;
  is_justif_preca: BooleanNumber;
  is_autres: BooleanNumber;
  is_aft: BooleanNumber;
  is_contrat_distrib: BooleanNumber;
  is_conv_tepcv: BooleanNumber;
  is_conv_regroupement: BooleanNumber;
  is_liste_depenses: BooleanNumber;
  is_attestation_tepcv: BooleanNumber;
  is_attestation_collectivite: BooleanNumber;
  is_etat_recapitulatif: BooleanNumber;
  is_ah: BooleanNumber;
  is_horodatage: BooleanNumber;
  is_subrogation: BooleanNumber;
  statut: number;
  litige?: BooleanNumber;
}

export interface File extends SimpleFile {}

export interface FileFull extends SimpleFile {}

export interface Files {
  [index: number]: File;
}

export interface MOA {
  moa_prenom: string;
  moa_nom: string;
  moa_fonction: string;
  moa_rue: string;
  moa_rue2: string;
  moa_cp: string;
  moa_ville: string;
  moa_denomination: string;
}

export interface FolderMOAString {
  moa_civilite: string;
  moa_nom: string;
  moa_prenom: string;
  moa_rue: string;
  moa_rue2: string;
  moa_cp: string;
  moa_ville: string;
  moa_pays: string;
  moa_tel: string;
  moa_tel_2: string;
  moa_mobile: string;
  moa_email: string;
  moa_bic: string;
  moa_iban: string;
  moa_siret: string;
  moa_denomination: string;
  moa_fax: string;
  moa_commentaire: string;
  moa_individu_email: string;
  moa_contact: string;
  moa_contact_mobile: string;
  moa_raison_sociale_siege: string;
  moa_siren_siege: string;
  moa_adresse_siege: string;
  moa_cp_siege: string;
  moa_ville_siege: string;
  moa_fonction: string | null;
  moa_individu_fonction: string | null;
  moa_beneficiaire_role: string | null;
}

export interface SimpleFolder extends FolderMOAString {
  id_dp_operation: number;
  id_dossierprime: number;
  code_operation: string;
  moa?: MOA[];
  moe?: {}[];
  travaux?: {}[];
  moa_est_societe: 0;
  moa_is_syndic: BooleanNumber;
  moa_no_siret: BooleanNumber;
  moa_categorie_menage: number;
  moa_nombre_menage: number;
  moa_nombre_personne: number;
  moa_remp_preca: number;
  moe_denomination: string;
  moe_siret: string;
  moe_tel: string;
  moe_fax: string;
  moe_rue: string;
  moe_cp: string;
  moe_ville: string;
  moe_role: number;
  moe_commentaire: string;
  moe_individu_email: string;
  moe_contact: string;
  moe_contact_prenom: string;
  moe_contact_mobile: string;
  moe_contact_fonction: string;
  moe_bic: string;
  moe_iban: string;
  adresse_travaux_rue: string;
  adresse_travaux_nomsite: string;
  adresse_travaux_rue2: string;
  adresse_travaux_cp: string;
  adresse_travaux_ville: string;
  date_debut_travaux: string | null;
  date_fin_travaux: string | null;
  /* code_operation: "BAR-TH-106"
  date_reception: "2018-06-11"
  delai_instruction?: "331 jours de retard"
  id_dossierprime: 19925
  id_dp_operation: 60839
  is_avant_projet: 0
  label_public: "Dossier Incomplet"
  moa_denomination: ""
  moa_nom: "JAUD"
  moa_prenom: "Jean-Marie"
  statut_operation: 13 */
}

export interface Folder extends SimpleFolder {
  documents?: number[];
}

export interface FolderFull extends SimpleFolder {
  documents?: FileFull[];
}

export interface Folders {
  [index: number]: File;
}

export interface CheckPointCategory {
  id_point_controle_categorie: number;
  code_categorie: string;
  nom_categorie: string | null;
}

export interface CheckPointCategories {
  [index: number]: CheckPointCategory;
}

export interface CheckPoint {
  id_point_controle: number;
  id_categorie: number;
  nom: string;
  id_penalite: BooleanNumber;
  automatique: BooleanNumber;
  is_controle_file: BooleanNumber;
  pivot: {
    id_dp_operation: number;
    id_point_controle: number;
    valide: BooleanNumber;
    id_dp_file?: number;
  };
  pointcontrolcategories: CheckPointCategory;
}

export interface CheckPoints {
  [index: number]: CheckPoint;
}

export interface OperationSimple {
  id_dp_operation: number;
  id_dossierprime: number;
  id_operation: number;
  code_operation: string;
  statut?: {
    code_statut?: number;
    label_public?: string;
    id_statut?: number;
    label_prive?: string;
  };
  moderemuneration: {
    id_remuneration: number;
    delai_instruction: string;
  };
}

export interface Operation extends OperationSimple {
  dossierprime?: number;
  dossierprimefile?: [number];
  point_controles: [number];
}

export interface OperationFull extends OperationSimple {
  dossierprime?: FolderFull;
  dossierprimefile?: [FileFull];
  point_controles: [CheckPoint];
}

export interface Operations {
  [index: number]: Operation;
}

export interface Entities {
  files: Files;
  folders: Folders;
  checkPoints: CheckPoints;
  checkPointCategories: CheckPointCategories;
  operations: Operations;
  users: Users;
  comments: Comments;
}

export interface Normalized {
  entities: Entities;
}

export interface FilesFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface FilesFolcerCheckPointLoaded {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADED;
  idDpFile: number;
  statusCode: number | null;
}

export interface FileLitigeLoaded {
  type: typeof FOLDER_FILE_LITIGE_LOADED;
  idDpFile: number;
  statusCode: number | null;
}

export interface FilesLogoutAction {
  type: typeof LOGOUT;
}

export type FilesActions = | FilesFolderLoadedAction
| FilesFolcerCheckPointLoaded
| FileLitigeLoaded
| FilesLogoutAction;

export interface FoldersFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface FoldersListLoadedAction {
  type: typeof LIST_LOADED;
  normalized: Normalized;
}

export interface FoldersLogoutAction {
  type: typeof LOGOUT;
}

export interface FoldersUpdateMoaLoaded {
  type: typeof FOLDER_UPDATE_MOA_LOADED;
  id_dossierprime: number;
  values: { [index: string]: string };
}

export type FoldersActions = | FoldersFolderLoadedAction
| FoldersLogoutAction
| FoldersUpdateMoaLoaded
| FoldersListLoadedAction;

export interface CheckPointsFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface CheckPointsListLoadedAction {
  type: typeof LIST_LOADED;
  normalized: Normalized;
}

export interface CheckPointsLogoutAction {
  type: typeof LOGOUT;
}

export interface CheckPointsFolderUpdateCheckpointLoadingAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADING;
  checkPointId: number;
}

export interface CheckPointsFolderUpdateChekpointLoadedAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_LOADED;
  folderId: number;
  checkPointId: number;
}

export interface CheckPointsFolderUpdateCheckpointErrorAction {
  type: typeof FOLDER_UPDATE_CHECK_POINT_ERROR;
  checkPointId: number;
  preValue: BooleanNumber;
}

export type CheckPointsActions = | CheckPointsFolderUpdateCheckpointLoadingAction
| CheckPointsFolderUpdateChekpointLoadedAction
| CheckPointsFolderUpdateCheckpointErrorAction
| CheckPointsFolderLoadedAction
| CheckPointsListLoadedAction
| CheckPointsLogoutAction;

export interface CheckPointCategoriesFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}

export interface CheckPointCategoriesLogoutAction {
  type: typeof LOGOUT;
}

export type CheckPointCategoriesActions = | CheckPointCategoriesFolderLoadedAction
| CheckPointCategoriesLogoutAction;

export interface OperationsFolderLoadedAction {
  type: typeof FOLDER_LOADED;
  normalized: Normalized;
}
export interface OperationsListLoadedAction {
  type: typeof LIST_LOADED;
  normalized: Normalized;
}

export interface OperationsLogoutAction {
  type: typeof LOGOUT;
}

export type OperationsActions = | OperationsFolderLoadedAction
| OperationsListLoadedAction
| OperationsLogoutAction;

export interface UsersCommentsListLoadedAction {
  type: typeof COMMENTS_LIST_LOADED;
  normalized: Normalized;
}

export type UsersActions = UsersCommentsListLoadedAction;

export interface CommentsCommentsListLoadedAction {
  type: typeof COMMENTS_LIST_LOADED;
  normalized: Normalized;
}

export type CommentsActions = CommentsCommentsListLoadedAction;
