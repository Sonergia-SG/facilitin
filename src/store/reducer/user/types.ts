import {
  USER_ADD_TOKEN,
  LOGOUT,
  USER_INFOS_LOADING,
  USER_INFOS_LOADED,
  USER_INFOS_ERROR,
  INIT,
} from '../../types';

export type UserFonction = 'instructeur_initial' | 'instructeur_suivi' | '';

export interface UserInfos {
  nom: string;
  prenom: string;
  identifiant: string;
  id_groupe: number;
  fonction: UserFonction;
}

export interface UserState {
  apiKey: string | null;
  infosLoading: boolean;
  user: UserInfos | null;
}

export interface UserInit {
  type: typeof INIT;
}

export interface UserAddTokenAction {
  type: typeof USER_ADD_TOKEN;
  apiKey: string | null;
}

export interface UserInfosLoading {
  type: typeof USER_INFOS_LOADING;
}

export interface UserInfosLoaded {
  type: typeof USER_INFOS_LOADED;
  user: UserInfos;
}

export interface UserInfosError {
  type: typeof USER_INFOS_ERROR;
}

export interface UserLogoutAction {
  type: typeof LOGOUT;
}

export type UserActionTypes = | UserInit
  | UserAddTokenAction
  | UserLogoutAction
  | UserInfosLoading
  | UserInfosLoaded
  | UserInfosError;
