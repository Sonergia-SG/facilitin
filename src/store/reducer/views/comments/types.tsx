import {
  INIT,
  COMMENTS_LIST_LOADING,
  COMMENTS_LIST_LOADED,
  COMMENTS_LIST_ERROR,
  UPDATE_NEW_COMMENT_MESSAGE,
  COMMENTS_POST_COMMENT_LOADING,
  COMMENTS_POST_COMMENT_LOADED,
  COMMENTS_POST_COMMENT_ERROR,
} from '../../../types';
import { CommentsLoadedNormalized } from '../../../actions/views/comments';

export interface CommentsByFoldersPending {
  newMessage: string;
  loading: boolean;
  postLoading: boolean;
}

export interface CommentsByFolders {
  pending: CommentsByFoldersPending;
  comments: Array<number>;
}

export interface CommentsState {
  byFolders: {
    [index: number]: CommentsByFolders;
  };
}

interface CommentsIniAction {
  type: typeof INIT;
}

export interface CommentsListLoadingAction {
  type: typeof COMMENTS_LIST_LOADING;
  idDpFolder: number;
}

export interface CommentsListLoadedAction {
  type: typeof COMMENTS_LIST_LOADED;
  idDpFolder: number;
  normalized: CommentsLoadedNormalized;
}

export interface CommentsListErrorAction {
  type: typeof COMMENTS_LIST_ERROR;
  idDpFolder: number;
}

export interface CommentsUpdateNewMessage {
  type: typeof UPDATE_NEW_COMMENT_MESSAGE;
  message: string;
  idDpFolder: number;
}

export interface CommentsPostCommentLoading {
  type: typeof COMMENTS_POST_COMMENT_LOADING;
  idDpFolder: number;
}

export interface CommentsPostCommentLoaded {
  type: typeof COMMENTS_POST_COMMENT_LOADED;
  idDpFolder: number;
}

export interface CommentsPostCommentError {
  type: typeof COMMENTS_POST_COMMENT_ERROR;
  idDpFolder: number;
}

export type CommentsAction = | CommentsIniAction
| CommentsListLoadingAction
| CommentsListLoadedAction
| CommentsListErrorAction
| CommentsUpdateNewMessage
| CommentsPostCommentLoading
| CommentsPostCommentLoaded
| CommentsPostCommentError;
