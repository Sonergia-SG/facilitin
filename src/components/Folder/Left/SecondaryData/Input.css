.Input-Label {
  margin-bottom: 5px;
  margin-top: 0;
  font-size: 13px;
  font-weight: 700;
  color: #767676;
}

.Input {
  border: 1px solid #e2e2e4;
  box-shadow: none;
  outline: none;
  color: #c2c2c2;
  transition: bakground-color 0.2s ease;
}

.Input:not(.Input-Error-Msg):focus {
  border-color: #1fb5ad;
  box-shadow: none;
}

.Input:disabled {
  background-color: #eee;
}

.Input-Error-Msg {
  color: #fa8564;
  margin-top: 3px;
  font-size: 12px;
  text-align: right;
  min-height: 18px;
}
