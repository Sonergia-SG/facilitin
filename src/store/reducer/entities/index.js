import { LIST_LOADED } from '../../types';

const initialState = {
  folders: {},
  documents: {},
  moa: {},
  moe: {},
  travaux: {},
};

const entities = (state = initialState, action) => {
  switch (action.type) {
    case LIST_LOADED:
      return {
        ...state,
        folders: {
          ...state.folders,
          ...action.normalized.entities.folders,
        },
      };
    default:
      return state;
  }
};

export default entities;
