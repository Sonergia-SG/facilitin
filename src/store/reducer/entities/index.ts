import { combineReducers } from 'redux';

import folders from './folders';
import checkPoints from './checkPoints';
import files from './files';
import moa from './moa';
import moe from './moe';
import travaux from './travaux';

export default combineReducers({
  folders,
  moa,
  moe,
  travaux,
  checkPoints,
  files,
});
