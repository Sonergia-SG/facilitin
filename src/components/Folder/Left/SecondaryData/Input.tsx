import React, { ChangeEvent, Fragment } from 'react';
import idx from 'idx';

import { folderUpdateMoaValue } from '../../../../store/actions/views/folder';
import { FolderFull, FolderMOAString } from '../../../../store/reducer/entities/types';
import { FolderPendingItem } from '../../../../store/reducer/views/folder/types';

interface Props {
  idDpOperation: number;
  valueKey: keyof FolderMOAString;
  label: string;
  disabled: boolean;
  dossierprime: FolderFull;
  pending?: FolderPendingItem;
  update: typeof folderUpdateMoaValue;
}

const Input = ({
  pending,
  valueKey,
  dossierprime,
  label,
  disabled,
  update,
  idDpOperation,
}: Props) => {
  const value = idx(pending, _ => _.moa[valueKey]);
  const originalValue = dossierprime[valueKey];
  const cleanOriginalValue = originalValue === null ? undefined : originalValue;

  const isEdited = value !== undefined;

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    update(idDpOperation, valueKey, e.target.value);
  };

  return (
    <Fragment>
      <label htmlFor={valueKey}>{label}</label>
      <input
        type="text"
        name={valueKey}
        placeholder={label}
        className={`input${isEdited ? ' is-info' : ''}`}
        disabled={disabled}
        value={typeof value === 'string' ? value : cleanOriginalValue}
        onChange={handleChange}
      />
    </Fragment>
  );
};

export default Input;
