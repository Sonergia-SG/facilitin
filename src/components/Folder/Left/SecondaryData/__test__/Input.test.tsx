import React from 'react';
import { shallow } from 'enzyme';

import Input from '../Input';
import { BooleanNumber } from '../../../../../store/reducer/entities/types';

const defaultDossierPrime = {
  id_dp_operation: 123,
  id_dossierprime: 123,
  code_operation: '03ER',
  moa_est_societe: 0,
  moa_is_syndic: 0 as BooleanNumber,
  moa_no_siret: 0 as BooleanNumber,
  moa_categorie_menage: 1,
  moa_nombre_menage: 1,
  moa_nombre_personne: 2,
  moa_remp_preca: 3,
  moe_role: 3,
  moa_civilite: '',
  moa_nom: '',
  moa_prenom: '',
  moa_rue: '',
  moa_rue2: '',
  moa_cp: '',
  moa_ville: '',
  moa_pays: '',
  moa_tel: '',
  moa_tel_2: '',
  moa_mobile: '',
  moa_email: '',
  moa_bic: '',
  moa_iban: '',
  moa_siret: '',
  moa_denomination: '',
  moa_fax: '',
  moa_commentaire: '',
  moa_individu_email: '',
  moa_contact: '',
  moa_contact_mobile: '',
  moa_raison_sociale_siege: '',
  moa_siren_siege: '',
  moa_adresse_siege: '',
  moa_cp_siege: '',
  moa_ville_siege: '',
  moa_fonction: null,
  moa_individu_fonction: null,
  moa_beneficiaire_role: null,
  moe_denomination: '',
  moe_siret: '',
  moe_tel: '',
  moe_fax: '',
  moe_rue: '',
  moe_cp: '',
  moe_ville: '',
  moe_commentaire: '',
  moe_individu_email: '',
  moe_contact: '',
  moe_contact_prenom: '',
  moe_contact_mobile: '',
  moe_contact_fonction: '',
  moe_bic: '',
  moe_iban: '',
  adresse_travaux_rue: '',
  adresse_travaux_nomsite: '',
  adresse_travaux_rue2: '',
  adresse_travaux_cp: '',
  adresse_travaux_ville: '',
  date_debut_travaux: null,
  date_fin_travaux: null,
};

describe('SecondaryData Input', () => {
  it('render rigth Simple case', () => {
    const wrapper = shallow(
      <Input
        key="azer"
        label="test :"
        valueKey="moa_nom"
        value="Sonergia"
        type="text"
        rules={undefined}
        idDpOperation={1}
        disabled={false}
        dossierprime={defaultDossierPrime}
        pending={undefined}
        pendingKey="moa"
        update={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render rigth Simple case - pending value', () => {
    const wrapper = shallow(
      <Input
        key="azer"
        label="test :"
        valueKey="moa_nom"
        value="Sonergia"
        type="text"
        rules={undefined}
        idDpOperation={1}
        disabled={false}
        dossierprime={defaultDossierPrime}
        pending={{ moa: { moa_nom: 'So' } }}
        pendingKey="moa"
        update={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render rigth Simple case - disabled', () => {
    const wrapper = shallow(
      <Input
        key="azer"
        label="test :"
        valueKey="moa_nom"
        value="Sonergia"
        type="text"
        rules={undefined}
        idDpOperation={1}
        disabled
        dossierprime={defaultDossierPrime}
        pending={undefined}
        pendingKey="moa"
        update={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render rigth with valid phone', () => {
    const wrapper = shallow(
      <Input
        key="azer"
        label="tel :"
        valueKey="moa_tel"
        value="06.06.06.06.06"
        type="text"
        rules={{ format: 'phone' }}
        idDpOperation={1}
        disabled={false}
        dossierprime={defaultDossierPrime}
        pending={undefined}
        pendingKey="moa"
        update={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render rigth with invalid phone', () => {
    const wrapper = shallow(
      <Input
        key="azer"
        label="tel :"
        valueKey="moa_tel"
        value="06.06.06.06"
        type="text"
        rules={{ format: 'phone' }}
        idDpOperation={1}
        disabled={false}
        dossierprime={defaultDossierPrime}
        pending={undefined}
        pendingKey="moa"
        update={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render rigth with invalid bic', () => {
    const wrapper = shallow(
      <Input
        key="azer"
        label="tel :"
        valueKey="moa_tel"
        value="06.06.06.06"
        type="text"
        rules={{ format: 'bic' }}
        idDpOperation={1}
        disabled={false}
        dossierprime={defaultDossierPrime}
        pending={undefined}
        pendingKey="moa"
        update={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render rigth with invalid iban', () => {
    const wrapper = shallow(
      <Input
        key="azer"
        label="tel :"
        valueKey="moa_tel"
        value="06.06.06.06"
        type="text"
        rules={{ format: 'iban' }}
        idDpOperation={1}
        disabled={false}
        dossierprime={defaultDossierPrime}
        pending={undefined}
        pendingKey="moa"
        update={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render rigth with invalid siret', () => {
    const wrapper = shallow(
      <Input
        key="azer"
        label="tel :"
        valueKey="moa_tel"
        value="78953453400018"
        type="text"
        rules={{ format: 'siret' }}
        idDpOperation={1}
        disabled={false}
        dossierprime={defaultDossierPrime}
        pending={undefined}
        pendingKey="moa"
        update={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render rigth with valid siret', () => {
    const wrapper = shallow(
      <Input
        key="azer"
        label="tel :"
        valueKey="moa_tel"
        value="78953453400019"
        type="text"
        rules={{ format: 'siret' }}
        idDpOperation={1}
        disabled={false}
        dossierprime={defaultDossierPrime}
        pending={undefined}
        pendingKey="moa"
        update={() => {}}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
