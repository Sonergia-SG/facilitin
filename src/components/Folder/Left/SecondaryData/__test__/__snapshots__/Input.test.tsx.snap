// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryData Input render rigth Simple case - disabled 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_nom"
  >
    test :
  </label>
  <input
    className="Input input"
    disabled={true}
    name="moa_nom"
    onChange={[Function]}
    placeholder="test :"
    type="text"
    value="Sonergia"
  />
  <p
    className="Input-Error-Msg"
  />
</Fragment>
`;

exports[`SecondaryData Input render rigth Simple case - pending value 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_nom"
  >
    test :
  </label>
  <input
    className="Input input is-info"
    disabled={false}
    name="moa_nom"
    onChange={[Function]}
    placeholder="test :"
    type="text"
    value="So"
  />
  <p
    className="Input-Error-Msg"
  />
</Fragment>
`;

exports[`SecondaryData Input render rigth Simple case 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_nom"
  >
    test :
  </label>
  <input
    className="Input input"
    disabled={false}
    name="moa_nom"
    onChange={[Function]}
    placeholder="test :"
    type="text"
    value="Sonergia"
  />
  <p
    className="Input-Error-Msg"
  />
</Fragment>
`;

exports[`SecondaryData Input render rigth with invalid bic 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_tel"
  >
    tel :
  </label>
  <input
    className="Input input is-danger"
    disabled={false}
    name="moa_tel"
    onChange={[Function]}
    placeholder="tel :"
    type="text"
    value="06.06.06.06"
  />
  <p
    className="Input-Error-Msg"
  >
    BIC invalide
  </p>
</Fragment>
`;

exports[`SecondaryData Input render rigth with invalid email 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_email"
  >
    email :
  </label>
  <input
    className="Input input is-danger"
    disabled={false}
    name="moa_email"
    onChange={[Function]}
    placeholder="email :"
    type="text"
    value="0421"
  />
  <p
    className="Input-Error-Msg"
  >
    Email invalide
  </p>
</Fragment>
`;

exports[`SecondaryData Input render rigth with invalid fiscal number 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_tel"
  >
    tel :
  </label>
  <input
    className="Input input is-danger"
    disabled={false}
    name="moa_tel"
    onChange={[Function]}
    placeholder="tel :"
    type="text"
    value="78953453azez"
  />
  <p
    className="Input-Error-Msg"
  >
    Num√©ro fiscale invalide
  </p>
</Fragment>
`;

exports[`SecondaryData Input render rigth with invalid iban 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_tel"
  >
    tel :
  </label>
  <input
    className="Input input is-danger"
    disabled={false}
    name="moa_tel"
    onChange={[Function]}
    placeholder="tel :"
    type="text"
    value="06.06.06.06"
  />
  <p
    className="Input-Error-Msg"
  >
    IBAN invalide
  </p>
</Fragment>
`;

exports[`SecondaryData Input render rigth with invalid phone 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_tel"
  >
    tel :
  </label>
  <input
    className="Input input is-danger"
    disabled={false}
    name="moa_tel"
    onChange={[Function]}
    placeholder="tel :"
    type="text"
    value="06.06.06.06"
  />
  <p
    className="Input-Error-Msg"
  >
    Format invalide (XX.XX.XX.XX.XX)
  </p>
</Fragment>
`;

exports[`SecondaryData Input render rigth with invalid postal code 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_cp"
  >
    code postal :
  </label>
  <input
    className="Input input is-danger"
    disabled={false}
    name="moa_cp"
    onChange={[Function]}
    placeholder="code postal :"
    type="text"
    value="0421"
  />
  <p
    className="Input-Error-Msg"
  >
    Code postal invalide
  </p>
</Fragment>
`;

exports[`SecondaryData Input render rigth with invalid siret 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_tel"
  >
    tel :
  </label>
  <input
    className="Input input is-danger"
    disabled={false}
    name="moa_tel"
    onChange={[Function]}
    placeholder="tel :"
    type="text"
    value="78953453400018"
  />
  <p
    className="Input-Error-Msg"
  >
    SIRET invalide
  </p>
</Fragment>
`;

exports[`SecondaryData Input render rigth with valid email 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_email"
  >
    email :
  </label>
  <input
    className="Input input"
    disabled={false}
    name="moa_email"
    onChange={[Function]}
    placeholder="email :"
    type="text"
    value="igor.eschalier@sonergia.fr"
  />
  <p
    className="Input-Error-Msg"
  />
</Fragment>
`;

exports[`SecondaryData Input render rigth with valid fiscal number 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_tel"
  >
    tel :
  </label>
  <input
    className="Input input"
    disabled={false}
    name="moa_tel"
    onChange={[Function]}
    placeholder="tel :"
    type="text"
    value="78 95 345 340 001"
  />
  <p
    className="Input-Error-Msg"
  />
</Fragment>
`;

exports[`SecondaryData Input render rigth with valid phone 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_tel"
  >
    tel :
  </label>
  <input
    className="Input input"
    disabled={false}
    name="moa_tel"
    onChange={[Function]}
    placeholder="tel :"
    type="text"
    value="06.06.06.06.06"
  />
  <p
    className="Input-Error-Msg"
  />
</Fragment>
`;

exports[`SecondaryData Input render rigth with valid postal code 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_cp"
  >
    code postal :
  </label>
  <input
    className="Input input"
    disabled={false}
    name="moa_cp"
    onChange={[Function]}
    placeholder="code postal :"
    type="text"
    value="04210"
  />
  <p
    className="Input-Error-Msg"
  />
</Fragment>
`;

exports[`SecondaryData Input render rigth with valid siret 1`] = `
<Fragment>
  <label
    className="Input-Label"
    htmlFor="moa_tel"
  >
    tel :
  </label>
  <input
    className="Input input"
    disabled={false}
    name="moa_tel"
    onChange={[Function]}
    placeholder="tel :"
    type="text"
    value="78953453400019"
  />
  <p
    className="Input-Error-Msg"
  />
</Fragment>
`;
